login-workflow/src/pages/create-account/steps/account-details/account-details.component.ts

```diff
@@ -31,6 +31,7 @@ import { PxbAuthTranslations } from '../../../../translations/auth-translations'
                                 id="pxb-first"
                                 name="first"
                                 matInput
+                                [maxLength]="firstNameMaxLength"
```

Comment:
We should be able to conditionally add this attribute right? Like if customFirstNameRequirements is undefined or customFirstNameRequirements.maxLength is undefined we don't specify this attribute at all.

Reply:
This is the reason why I defaulted to the 1000 character limit; I don't know if we can conditionally apply the directive without having to render 2 elements - one with the maxLength applied and the other without.  

The maxLength directive doesn't accept an input that'll disable it.

Ideally we would do this:

` [maxLength]="firstNameMaxLength ? firstNameMaxLength  : undefined"`

But the above case, the undefined === 0 and disables the input field.

Reply:
I'm refactoring this to not use the directive, I'll use FormControl Validators instead.

Reply:
FormContorl Validator.maxLength does not prevent a user from typing in more than the character limit.

Reply:
Did you try `null` instead of undefined? https://codippa.com/how-to-add-an-attribute-to-an-element-conditionally-in-angular/

Reply:
This is exactly what we needed!